project_key/models/project_project.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2017 - 2018 Modoolar <info@modoolar.com>
+# License LGPLv3.0 or later (https://www.gnu.org/licenses/lgpl-3.0.en.html).
+
+from odoo import models, fields, api, _
+from odoo.osv import expression
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+
+    task_key_sequence_id = fields.Many2one(
+        comodel_name='ir.sequence',
+        string='Key Sequence',
+        ondelete="restrict",
+    )
+
+    key = fields.Char(
+        string='Key',
+        size=10,
+        required=True,
+        index=True,
+    )
+
+    _sql_constraints = [
+        ("project_key_unique", "UNIQUE(key)", "Project key must be unique!")
+    ]
+
+    @api.multi
+    @api.onchange('name')
+    def _onchange_project_name(self):
+        for rec in self:
+            if rec.name:
+                rec.key = self.generate_project_key(rec.name)
+            else:
+                rec.key = ''
+
+    @api.model
+    def create(self, vals):
+        if 'key' not in vals:
+            if vals['name']:
```

Comment:
`name` is a required field. If it can be falsey, maybe it can also be the case that it is not in `vals` so you might want to use `vals.get('name')`. If it cannot, do away with the condition entirely.

Reply:
But can it be falsey?

Reply:
I see :D

Reply:
Fixed.